packages/realm/src/TypeHelpers.ts

```diff
@@ -98,6 +102,12 @@ export function mixedToBinding(realm: binding.Realm, value: unknown): binding.Mi
     throw new Error(`Using a ${value.constructor.name} as Mixed value, is not yet supported`);
   } else if (Array.isArray(value)) {
     throw new TypeError("A mixed property cannot contain an array of values.");
+  } else if (isGeoCircle(value)) {
+    return binding.Geospatial.fromCircle(value);
+  } else if (isGeoBox(value)) {
+    return binding.Geospatial.fromBox(value);
+  } else if (isGeoPolygon(value)) {
+    return binding.Geospatial.fromPolygon(value);
```

Comment:
For now, this feels a bit like the wrong place for this. I don't think we want to accept geo stuff every place where we accept `Mixed`. Instead, lets add this logic specifically in the query args API.

Also, I wouldn't frame it as an independent set of if/else if `ifFoo` checks with their own duplicated checks. Instead, lets properly define what we will look for when making the decision.

I suggest something like:

```ts
if (typeof(value) !== 'object') return Mixed()
if (Array.isArray(val)) return Point()
if ('type' in value) {
    if (value.type === 'Point') return Point()
    if (value.type === 'Polygon' return Polygon()
}
if ('latitude' in value) return Point()
if ('topRight' in value) return GeoBox()
if ('center' in value) return GeoCircle()
```

Alternatively, we could use classes with constructors and require the user to do `new GeoPoint(flexibleArg)` and just have a chain of `if (value instanceof GeoPoint)` style checks.

Reply:
Regarding where to put this code... It seems that `mixedToBinding` is used anyway only for the query arguments. 
If at some point this method is going to be used also for something else, then we could add a boolean input that would skip the geospatial discriminators

Reply:
Regarding the duplicated checks... I think the only duplicated one for now is `typeof value === object and value !== null`, as we can't have GeoPoints as arguments.
I will move those two checks outside of the type predicates.